packages/multi-test/src/test_helpers/contracts/error.rs

```diff
@@ -0,0 +1,44 @@
+use std::fmt;
+
+use cosmwasm_std::{Binary, Deps, DepsMut, Empty, Env, MessageInfo, Response, StdError};
+use schemars::JsonSchema;
+
+use crate::{test_helpers::EmptyMsg, Contract, ContractWrapper};
+
+fn instantiate(
+    _deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    _msg: EmptyMsg,
+) -> Result<Response, StdError> {
+    Err(StdError::generic_err("Init failed"))
+}
+
+fn execute(
+    _deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    _msg: EmptyMsg,
+) -> Result<Response, StdError> {
+    Err(StdError::generic_err("Handle failed"))
+}
+
+fn query(_deps: Deps, _env: Env, _msg: EmptyMsg) -> Result<Binary, StdError> {
+    Err(StdError::generic_err("Query failed"))
+}
+
+pub fn contract() -> Box<dyn Contract<Empty>> {
+    let contract: ContractWrapper<_, _, _, _, _, _, _, _, _> =
+        ContractWrapper::new(execute, instantiate, query);
+    Box::new(contract)
+}
+
+#[allow(dead_code)]
```

Comment:
good for completeness

Reply:
I wonder if we could just merge with the above? It is less efficient to add a transform from Empty -> Empty but this is only for test code? Does smell funny though

Reply:
Actually I didn't want to make this for now just because I was still thinking about this, but I see literally no reason to split those functions.

Also while we are at this point - I see no reason to make those returning `Box<dyn Contract<C>>` instead of `impl Contract<C>` - it may always be boxed when needed. I see we are in general overusing dynamic dispatch - I find them not working so well with typesystem in general.

Reply:
>  I see we are in general overusing dynamic dispatch - I find them not working so well with typesystem in general.

I used to have 2-4 generics on almost every function. This was replaced with eg `&dyn Storage` and I found it made everything more readable and usable in general.

We can add in some more generics as you see fit, but I would like to keep the number relatively low.

Reply:
> I see no reason to make those returning Box<dyn Contract<C>> instead of impl Contract<C> 

You can make a commit doing that change, and if it makes code simpler, then let's use it.